datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -0,0 +1,538 @@
+        let optimized_plan = rule
+            .optimize(plan, &ExecutionProps::new())
+            .expect("failed to optimize plan");
+        let formatted_plan = format!("{:?}", optimized_plan);
+        assert_eq!(formatted_plan, expected);
+    }
+
+    #[test]
+    fn test_simplify_or_true() -> Result<()> {
+        let expr_a = binary_expr(col("c"), Operator::Or, lit(true));
+        let expr_b = binary_expr(lit(true), Operator::Or, col("c"));
+        let expected = lit(true);
+
+        assert_eq!(simplify(&expr_a), expected);
+        assert_eq!(simplify(&expr_b), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_or_false() -> Result<()> {
+        let expr_a = binary_expr(lit(false), Operator::Or, col("c"));
+        let expr_b = binary_expr(col("c"), Operator::Or, lit(false));
+        let expected = col("c");
+
+        assert_eq!(simplify(&expr_a), expected);
+        assert_eq!(simplify(&expr_b), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_or_same() -> Result<()> {
+        let expr = binary_expr(col("c"), Operator::Or, col("c"));
+        let expected = col("c");
+
+        assert_eq!(simplify(&expr), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_and_false() -> Result<()> {
+        let expr_a = binary_expr(lit(false), Operator::And, col("c"));
+        let expr_b = binary_expr(col("c"), Operator::And, lit(false));
+        let expected = lit(false);
+
+        assert_eq!(simplify(&expr_a), expected);
+        assert_eq!(simplify(&expr_b), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_and_same() -> Result<()> {
+        let expr = binary_expr(col("c"), Operator::And, col("c"));
+        let expected = col("c");
+
+        assert_eq!(simplify(&expr), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_and_true() -> Result<()> {
+        let expr_a = binary_expr(lit(true), Operator::And, col("c"));
+        let expr_b = binary_expr(col("c"), Operator::And, lit(true));
+        let expected = col("c");
+
+        assert_eq!(simplify(&expr_a), expected);
+        assert_eq!(simplify(&expr_b), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_multiply_by_one() -> Result<()> {
+        let expr_a = binary_expr(col("c"), Operator::Multiply, lit(1));
+        let expr_b = binary_expr(lit(1), Operator::Multiply, col("c"));
+        let expected = col("c");
+
+        assert_eq!(simplify(&expr_a), expected);
+        assert_eq!(simplify(&expr_b), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_divide_by_one() -> Result<()> {
+        let expr = binary_expr(col("c"), Operator::Divide, lit(1));
+        let expected = col("c");
+
+        assert_eq!(simplify(&expr), expected);
+        Ok(())
+    }
+
+    #[test]
+    fn test_simplify_divide_by_same() -> Result<()> {
+        let expr = binary_expr(col("c"), Operator::Divide, col("c"));
```

Comment:
I wonder if we care that in some cases (where c has a `0`) this simplification will avoid a runtime error when it would have generated one without the optimization pass

I personally think it is ok, but wanted to mention it